alacritty_terminal/src/term/mod.rs

```diff
@@ -1368,9 +1319,71 @@ impl Term {
     pub fn clipboard(&mut self) -> &mut Clipboard {
         &mut self.clipboard
     }
+
+    pub fn urls(&self) -> Vec<Url> {
+        let display_offset = self.grid.display_offset();
+        let num_cols = self.grid.num_cols().0;
+        let last_col = Column(num_cols - 1);
```

Comment:
You don't need `-1` here, you have a big chance of skipping last `WRAPLINE` cell (in fish shell you're skipping it).

Reply:
Maybe, I'm wrong here, but there's something broken with multiline urls, but there's definitely something odd with all of this story. For e.g. if I create a long line in vim with url, it wouldn't be considered as multiline one (I don't have any any vim decorations on the left), same here for the fish shell, if I write a url and move it to the right with space chars(to make it appear on multiple lines), it wouldn't be considered as multiline one, until I resize a terminal.

Reply:
We kinda want a `-1` for correctness here, similar to the issue in `url_to_string` though the iterators do not include the cell they're right above. Since cells are 0-based the last column has the index num_cols - 1.

It might just work to start 'outside' the actual grid, however that seems kinda troublesome and could actually lead to index out of bounds if things ever change in the future.

Reply:
Yeah, I’m just wondering what’s going wrong with long wrapped url in neovim/fish. Unfortunately, I can’t debug this one atm.

Reply:
I'll try to find a clean solution for fixing the iterators and I'll try to repro once that's fixed.

Reply:
I've resolved the mentioned issues in the latest version. I don't think there's anything we can do about long wrapped URLs in neovim, since these are actually broken by neovim with a hard linebreak at the end.